`setuptools` is not listed as a dependency

Importing Nornir in a Python 3.12 virtual environment fails because `setuptools` is not installed:

```
    from nornir import InitNornir
.venv/lib/python3.12/site-packages/nornir/__init__.py:1: in <module>
    import pkg_resources
E   ModuleNotFoundError: No module named 'pkg_resources'
```

In Python 3.12, `setuptools` is no longer installed in virtual environments (https://docs.python.org/3/whatsnew/3.12.html):

> [gh-95299](https://github.com/python/cpython/issues/95299): Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.

Manually installing `setuptools` works, but it should be listed as an explicit dependency of Nornir.

I'd make a PR, but I'm not sure what version constraint to use.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,6 +30,7 @@ python = ">=3.8,<4.0"
 "ruamel.yaml" = ">=0.17"
 # importlib.metadata has an API change in 3.10 and is no longer provisional
 importlib-metadata = { version = "^4", python = "<3.10" }
+setuptools = ">=42.0.0"
 
 [tool.poetry.group.dev.dependencies]
 decorator = "*"

